diff -urNp -x '*.orig' dbus-1.12.20.org/bus/session.conf.in dbus-1.12.20/bus/session.conf.in
--- dbus-1.12.20.org/bus/session.conf.in	2017-10-30 13:26:18.000000000 +0100
+++ dbus-1.12.20/bus/session.conf.in	2022-01-08 14:11:46.333563322 +0100
@@ -33,6 +33,10 @@
     <allow eavesdrop="true"/>
     <!-- Allow anyone to own anything -->
     <allow own="*"/>
+    <!-- Added by PLD -->
+    <!-- valid replies are always allowed -->
+    <allow send_requested_reply="true"/>
+    <allow receive_requested_reply="true"/>
   </policy>
 
   <!-- Include legacy configuration location -->
