--- dbus-0.60/gcj/org/freedesktop/dbus/Makefile.am.orig	2005-09-07 00:38:54.000000000 +0200
+++ dbus-0.60/gcj/org/freedesktop/dbus/Makefile.am	2006-01-23 21:34:19.412111750 +0100
@@ -23,7 +23,7 @@
 java_object_files = $(java_sources:.java=.lo)
 
 .class.h:
-	$(GCJH) $(basename $<)
+	$(GCJH) -I. $(basename $<)
 
 .java.class:
 	$(GCJ) -C $< -d ../../..
--- dbus-0.60/gcj/Makefile.am.orig	2006-01-23 21:36:39.364858250 +0100
+++ dbus-0.60/gcj/Makefile.am	2006-01-23 21:35:24.720193250 +0100
@@ -1,5 +1,5 @@
 SUBDIRS=org
-JAR=jar
+JAR=fastjar
 
 CLEANFILES=dbus-1.jar
 
@@ -14,6 +14,7 @@
 
 noinst_PROGRAMS = test-message
 
+AM_GCJFLAGS=-I.
 test_message_SOURCES = TestMessage.java
 test_message_LDFLAGS = --main=TestMessage
 test_message_LDADD = $(top_builddir)/gcj/org/freedesktop/dbus/libdbus-gcj-1.la
