--- ../BUILD/dbus-0.50/mono/Makefile.am.orig	2005-01-25 20:47:13.000000000 +0100
+++ ../BUILD/dbus-0.50/mono/Makefile.am	2005-12-03 16:00:44.000000000 +0100
@@ -57,14 +57,14 @@
 
 install-data-local:
 	@if test -n '$(TARGET)'; then                       \
-	  echo "$(GACUTIL) /i $(ASSEMBLY) /f /package dbus-sharp /gacdir $(DESTDIR)$(libdir)";    \
-	  $(GACUTIL) /i $(ASSEMBLY) /f /package dbus-sharp /gacdir $(libdir) /root $(DESTDIR)$(libdir) || exit 1; \
+	  echo "$(GACUTIL) /i $(ASSEMBLY) /f /package dbus-sharp /gacdir $(DESTDIR)$(prefix)/lib";    \
+	  $(GACUTIL) /i $(ASSEMBLY) /f /package dbus-sharp /gacdir $(libdir) /root $(DESTDIR)$(prefix)/lib || exit 1; \
 	fi
 
 uninstall-local:
 	@if test -n '$(TARGET)'; then                       \
-	  echo "$(GACUTIL) /u $(ASSEMBLY_NAME) /package dbus-sharp /gacdir $(DESTDIR)$(libdir)";  \
-	  $(GACUTIL) /u $(ASSEMBLY_NAME) /package dbus-sharp /gacdir $(libdir) /root $(DESTDIR)$(libdir) || exit 1;   \
+	  echo "$(GACUTIL) /u $(ASSEMBLY_NAME) /package dbus-sharp /gacdir $(DESTDIR)$(prefix)/lib";  \
+	  $(GACUTIL) /u $(ASSEMBLY_NAME) /package dbus-sharp /gacdir $(libdir) /root $(DESTDIR)$(prefix)/lib || exit 1;   \
 	fi
 
 EXTRA_DIST=			\
--- ../BUILD/dbus-0.50/dbus-sharp.pc.in.orig	2004-06-10 14:55:28.000000000 +0200
+++ ../BUILD/dbus-0.50/dbus-sharp.pc.in	2005-12-03 15:59:59.000000000 +0100
@@ -1,6 +1,6 @@
 prefix=@prefix@
 exec_prefix=@exec_prefix@
-libdir=@libdir@
+libdir=${prefix}/lib
 
 Name: DBus#
 Description: DBus# - D-BUS .NET Bindings
