--- dbus-0.35.2-old/python/dbus_bindings.pyx	2005-08-02 16:50:14.578553928 +0200
+++ dbus-0.35.2/python/dbus_bindings.pyx	2005-08-02 16:49:21.336647928 +0200
@@ -74,73 +74,73 @@
 
 class ObjectPath(str):
     def __init__(self, value):
-        str.__init__(value)
+        str.__init__(self, value)
 
 class ByteArray(str):
     def __init__(self, value):
-        str.__init__(value)
+        str.__init__(self, value)
 
 class Signature(str):
     def __init__(self, value):
-        str.__init__(value)
+        str.__init__(self, value)
 
 class Byte(int):
     def __init__(self, value):
-        int.__init__(value)
+        int.__init__(self, value)
 
 class Boolean(int):
     def __init__(self, value):
-        int.__init__(value)
+        int.__init__(self, value)
 
 class Int16(int):
     def __init__(self, value):
-        int.__init__(value)
+        int.__init__(self, value)
 
 class UInt16(int):
     def __init__(self, value):
         if value < 0:
             raise TypeError('Unsigned integers must not have a negitive value') 
-        int.__init__(value)
+        int.__init__(self, value)
 
 class Int32(int):
     def __init__(self, value):
-        int.__init__(value)
+        int.__init__(self, value)
 
 class UInt32(long):
     def __init__(self, value):
         if value < 0:
             raise TypeError('Unsigned integers must not have a negitive value') 
-        long.__init__(value)
+        long.__init__(self, value)
 
 class Int64(long):
     def __init__(self, value):
-        long.__init__(value)
+        long.__init__(self, value)
 
 class UInt64(long):
     def __init__(self, value):
         if value < 0:
             raise TypeError('Unsigned integers must not have a negitive value') 
-        long.__init__(value)
+        long.__init__(self, value)
 
 class Double(float):
     def __init__(self, value):
-        float.__init__(self, value)
+        float.__init__(self, self, value)
 
 class String(str):
     def __init__(self, value):
-        str.__init__(value)
+        str.__init__(self, value)
 
 class Array(list):
     def __init__(self, value):
-        list.__init__(value)
+        list.__init__(self, value)
 
 class Struct(tuple):
     def __init__(self, value):
-        tuple.__init__(value)
+        tuple.__init__(self, value)
 
 class Dictionary(dict):
     def __init__(self, value):
-        dict.__init__(value)
+        dict.__init__(self, value)
 
 #forward delcerations
 cdef class Message
