diff -urNp -x '*.orig' dbus-1.12.20.org/configure.ac dbus-1.12.20/configure.ac
--- dbus-1.12.20.org/configure.ac	2022-01-08 14:11:46.766881649 +0100
+++ dbus-1.12.20/configure.ac	2022-01-08 14:11:46.840212443 +0100
@@ -12,6 +12,7 @@ AC_CONFIG_AUX_DIR([build-aux])
 
 m4_pattern_forbid([^AX_(CHECK_ENABLE_DEBUG|CODE_COVERAGE|COMPILER_FLAGS|COMPILER_FLAGS_(CFLAGS|CXXFLAGS|LDFLAGS))\b],
   [Unexpanded AX_ macro found. Please install GNU autoconf-archive])
+m4_pattern_allow([AX_CHECK_GNU_MAKE_HEADLINE])
 
 AC_CANONICAL_HOST
 
