--- dbus-0.32/dbus/dbus-sysdeps-util.c.print_child_pid	2005-04-11 16:54:54.000000000 -0400
+++ dbus-0.32/dbus/dbus-sysdeps-util.c	2005-04-11 16:55:38.000000000 -0400
@@ -142,7 +142,7 @@
 	      return FALSE;
 	    }
 	  
-	  if (!_dbus_string_append_int (&pid, _dbus_getpid ()) ||
+	  if (!_dbus_string_append_int (&pid, child_pid) ||
 	      !_dbus_string_append (&pid, "\n"))
 	    {
 	      _dbus_string_free (&pid);
