--- dbus-1.2.10/bus/system.conf.in.orig	2008-12-18 22:10:55.000000000 +0100
+++ dbus-1.2.10/bus/system.conf.in	2008-12-26 02:04:26.330514668 +0100
@@ -55,6 +55,9 @@
     <allow send_type="signal"/>
     <allow send_requested_reply="true" send_type="method_return"/>
     <allow send_requested_reply="true" send_type="error"/>
+    <!-- Introspection and Peer methods are always allowed -->
+    <allow send_interface="org.freedesktop.DBus.Introspectable"/>
+    <allow send_interface="org.freedesktop.DBus.Peer"/>
 
     <!-- All messages may be received by default -->
     <allow receive_type="method_call"/>
--- dbus-1.2.10/test/data/valid-config-files-system/tmp-session-like-system.conf.in.orig	2008-12-19 20:39:55.000000000 +0100
+++ dbus-1.2.10/test/data/valid-config-files-system/tmp-session-like-system.conf.in	2008-12-26 09:42:41.950513830 +0100
@@ -24,6 +24,10 @@
     <deny own="*"/>
     <deny send_type="method_call"/>
 
+    <!-- Introspection and Peer methods are always allowed -->
+    <allow send_interface="org.freedesktop.DBus.Introspectable"/>
+    <allow send_interface="org.freedesktop.DBus.Peer"/>
+
     <!-- Signals and reply messages (method returns, errors) are allowed
          by default -->
     <allow send_type="signal"/>
