--- dbus-0.23.4/bus/session.conf.in.orig	2005-03-03 00:30:41.000000000 +0100
+++ dbus-0.23.4/bus/session.conf.in	2005-03-24 23:46:51.000000000 +0100
@@ -10,7 +10,7 @@
 
   <listen>unix:tmpdir=@DBUS_SESSION_SOCKET_DIR@</listen>
 
-  <servicedir>@EXPANDED_DATADIR@/dbus-1/services</servicedir>
+  <servicedir>@datadir@/dbus-1/services</servicedir>
 
   <policy context="default">
     <!-- Allow everything to be sent -->
@@ -19,6 +19,10 @@
     <allow eavesdrop="true"/>
     <!-- Allow anyone to own anything -->
     <allow own="*"/>
+    <!-- Added by PLD -->
+    <!-- valid replies are always allowed -->
+    <allow send_requested_reply="true"/>
+    <allow receive_requested_reply="true"/>
   </policy>
 
   <!-- This is included last so local configuration can override what's 
