diff -urN dbus-0.36.2.org/bus/session.conf.in dbus-0.36.2/bus/session.conf.in
--- dbus-0.36.2.org/bus/session.conf.in	2005-08-29 20:40:15.000000000 +0200
+++ dbus-0.36.2/bus/session.conf.in	2005-08-30 21:05:05.900734296 +0200
@@ -19,6 +19,10 @@
     <allow eavesdrop="true"/>
     <!-- Allow anyone to own anything -->
     <allow own="*"/>
+    <!-- Added by PLD -->
+    <!-- valid replies are always allowed -->
+    <allow send_requested_reply="true"/>
+    <allow receive_requested_reply="true"/>
   </policy>
 
   <!-- This is included last so local configuration can override what's 
