diff -aurN dbus-0.61.orig/gcj/Makefile.am dbus-0.61/gcj/Makefile.am
--- dbus-0.61.orig/gcj/Makefile.am	2006-02-10 21:57:10.000000000 +0100
+++ dbus-0.61/gcj/Makefile.am	2006-02-25 10:01:42.770512976 +0100
@@ -13,6 +13,7 @@
 
 noinst_PROGRAMS = test-message
 
+AM_GCJFLAGS=-I.
 test_message_SOURCES = TestMessage.java
 test_message_LDFLAGS = --main=TestMessage
 test_message_LDADD = $(top_builddir)/gcj/org/freedesktop/dbus/libdbus-gcj-1.la
diff -aurN dbus-0.61.orig/gcj/org/freedesktop/dbus/Makefile.am dbus-0.61/gcj/org/freedesktop/dbus/Makefile.am
--- dbus-0.61.orig/gcj/org/freedesktop/dbus/Makefile.am	2005-09-07 00:38:54.000000000 +0200
+++ dbus-0.61/gcj/org/freedesktop/dbus/Makefile.am	2006-02-25 10:00:45.082282920 +0100
@@ -23,7 +23,7 @@
 java_object_files = $(java_sources:.java=.lo)
 
 .class.h:
-	$(GCJH) $(basename $<)
+	$(GCJH) -I. $(basename $<)
 
 .java.class:
 	$(GCJ) -C $< -d ../../..
