--- configure.in.orig	2005-01-13 10:14:39.181278096 +0100
+++ configure.in	2005-01-13 10:16:51.891103136 +0100
@@ -193,10 +193,13 @@
 MONO_REQUIRED_VERSION=0.95
 AC_SUBST(MONO_REQUIRED_VERSION)
 PKG_CHECK_MODULES(MONO_DEPENDENCY, mono >= $MONO_REQUIRED_VERSION, have_mono=true, have_mono=false)
+PKG_CHECK_MODULES(MONO_DEPENDENCY, mint >= $MONO_REQUIRED_VERSION, have_mint=true, have_mint=false)
 
 if test "x$enable_mono" = "xyes"; then
 	if test "x$have_mono" = "xfalse"; then
-		AC_MSG_ERROR([Mono was not found])
+	    if test "x$have_mint" = "xfalse"; then
+		AC_MSG_ERROR([Mono or Mint was not found])
+	    fi
 	fi
 fi
 
@@ -213,10 +216,15 @@
 		LIB_SUFFIX=
 	fi
 else
+    if test "x$have_mint" = "xtrue"; then
+	AC_PATH_PROG(RUNTIME, mint, no)
+	AC_PATH_PROG(CSC, mcs, no)
+    else
 	AC_PATH_PROG(CSC, csc.exe, no)
 	RUNTIME=
 	LIB_PREFIX=
 	LIB_SUFFIX=.dylib
+    fi
 fi
 
 if test x$enable_mono = xyes; then
