diff -ur dbus-0.33-orig/bus/session.conf.in dbus-0.33/bus/session.conf.in
--- dbus-0.33-orig/bus/session.conf.in	2004-10-25 20:48:58.000000000 +0200
+++ dbus-0.33/bus/session.conf.in	2005-04-26 11:22:10.969586536 +0200
@@ -10,7 +10,7 @@
 
   <listen>unix:tmpdir=@DBUS_SESSION_SOCKET_DIR@</listen>
 
-  <servicedir>@EXPANDED_DATADIR@/dbus-1/services</servicedir>
+  <servicedir>@datadir@/dbus-1/services</servicedir>
 
   <policy context="default">
     <!-- Allow everything to be sent -->
@@ -21,6 +21,10 @@
     <allow own="*"/>
     <!-- Allow any user to connect -->
     <allow user="*"/>
+    <!-- Added by PLD -->
+    <!-- valid replies are always allowed -->
+    <allow send_requested_reply="true"/>
+    <allow receive_requested_reply="true"/>
   </policy>
 
   <!-- This is included last so local configuration can override what's 
