diff -Nuard dbus-0.22.orig/bus/session.conf.in dbus-0.22/bus/session.conf.in
--- bus/session.conf.in	2004-10-27 10:04:50.614822536 +0200
+++ bus/session.conf.in	2004-10-27 12:07:04.999826536 +0200
@@ -10,7 +10,7 @@
 
   <listen>unix:tmpdir=@DBUS_SESSION_SOCKET_DIR@</listen>
 
-  <servicedir>@EXPANDED_LIBDIR@/dbus-1.0/services</servicedir>
+  <servicedir>@datadir@/dbus-1/services</servicedir>
 
   <policy context="default">
     <!-- Allow everything to be sent -->
@@ -21,6 +21,10 @@
     <allow own="*"/>
     <!-- Allow any user to connect -->
     <allow user="*"/>
+    <!-- Added by PLD -->
+    <!-- valid replies are always allowed -->
+    <allow send_requested_reply="true"/>
+    <allow receive_requested_reply="true"/>
   </policy>
 
   <!-- This is included last so local configuration can override what's 
