Index: glib/dbus-gmain.c
===================================================================
RCS file: /cvs/dbus/dbus/glib/dbus-gmain.c,v
retrieving revision 1.38
diff -u -r1.38 dbus-gmain.c
--- glib/dbus-gmain.c	6 Apr 2005 17:36:47 -0000	1.38
+++ glib/dbus-gmain.c	12 Apr 2005 16:11:14 -0000
@@ -239,13 +239,7 @@
   handler = NULL;
 
   if (connection)
-    {
-      /* Dispatch messages */
-      while (dbus_connection_dispatch (connection) == DBUS_DISPATCH_DATA_REMAINS)
-        ;
-      
-      dbus_connection_unref (connection);
-    }
+    dbus_connection_unref (connection);
   
   return TRUE;
 }
